### ISSUE TITLE
Change the result type of the fold in classifySessionsOf?

### ISSUE THREAD

**Issue by harendra-kumar** [composewell/streamly#524]
```
classifySessionsOf
    :: (IsStream t, MonadAsync m, Ord k)
    => Double         -- ^ time window size
    -> (Int -> m Bool) -- ^ predicate to eject sessions on session count
    -> Fold m a (Either b b) -- ^ Fold to be applied to session events
    -> t m (k, a, AbsTime) -- ^ session key, data, timestamp
    -> t m (k, b)
```

The fold result is emitted in the output stream if the fold returns a `Left` result or if the time window ends. Paul suggests that the result type could be changed to `(b, Bool)` instead of `Either b b`. So it would be emitted in the output stream if the result is `(b, True)`.

However with terminating folds, we will not need the wrapper around the result. It can just be `Fold m a b`.

**Comment by adithyaov** [comment]
This is not breaking. `classifySessionsBy` is in an `Internal` module

**Comment by adithyaov** [comment]
Closed in #795
